src/sentry/models/email.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+from sentry.utils.http import absolute_uri
+
+
+class Email(Model):
+    __core__ = True
+
+    user = FlexibleForeignKey(settings.AUTH_USER_MODEL,
+                              unique=False,
+                              related_name='emails')
+    email = models.EmailField(_('email address'))
+    validation_hash = models.CharField(max_length=32)
+    date_hash_added = models.DateTimeField(default=timezone.now)
+    is_verified = models.BooleanField(
+        _('verified'), default=False,
+        help_text=_('Designates whether this user has confirmed their email.'))
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_email'
+        unique_together = (('user', 'email'),)
+
+    __repr__ = sane_repr('user_id', 'email')
+
+    def set_hash(self):
+        import hashlib
+        import random
+
+        self.date_hash_added = timezone.now()
+        self.validation_hash = hashlib.md5(str(random.randint(1, 10000000))).hexdigest()
```

Comment:
I did this because it is what we are doing in the reset password workflow: https://github.com/getsentry/sentry/blob/master/src/sentry/models/lostpasswordhash.py#L42

but I can definitely change it if uuid4().hex is better